src/benchmarklib/abstract_query_generator.hpp

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+#include <string>
+#include <vector>
+
+#include "strong_typedef.hpp"
+
+STRONG_TYPEDEF(size_t, QueryID);
+
+namespace opossum {
+
+// Query generators are used by benchmarks to generate SQL strings for the different benchmark queries. In their
+// simplest form, they return the same query each time `build_query` is invoked. For benchmarks like TPC-H, they can
+// also randomize parameters.
+
+class AbstractQueryGenerator {
+ public:
+  virtual ~AbstractQueryGenerator() = default;
+
+  // Generates a SQL string for a given (zero-indexed) query id
+  virtual std::string build_query(const QueryID query_id) = 0;
+
+  // Returns the names of the individual queries (e.g., "TPC-H 1")
+  const std::vector<std::string>& query_names() const;
+
+  // Returns the number of queries supported by the benchmark
+  QueryID::base_type num_available_queries() const;
+
+  // Returns the number of queries selected for execution
+  QueryID::base_type num_selected_queries() const;
```

Comment:
* Returning a ::base_type is not something we normally do. Why here? I'd just return size_t...
* In other places (Table::column_count(), ...) we use "_count" to indicate that the function returns a "count". Consistency would be nice here.

Reply:
ok